src/main/java/org/amoseman/budgetingbackend/password/Entropy.java

```diff
@@ -0,0 +1,33 @@
+package org.amoseman.budgetingbackend.password;
+
+import java.util.Optional;
+
+/**
+ * A class for calculating the bit entropy of a password.
+ */
+public class Entropy {
```

Comment:
This class name doesn't describe what it does.  This name tells me that it represents entropy, not that it calculates password character entropy.  Don't be afraid to use lengthy and descriptive class names.  EntropyCalculator, or even PasswordEntropyCalculator would be descriptive and immediately tell you what this single function class does.